src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -709,7 +709,7 @@ private void SetModelEventOnAnnotation()
         /// </summary>
         /// <param name="modelGuid">The model unique identifier.</param>
         /// <returns></returns>
-        private ModelBase annotationModel_GetModelBase(Guid modelGuid)
+        public ModelBase annotationModel_GetModelBase(Guid modelGuid)
```

Comment:
This event is registered on Workspace. So when you add a new annotation (be it a new annotation, Undo, paste or redo), this event has to be registered with that. Since the Paste() is on DynamModel, I had this set to public so that this event can be registered from Dynamomodel.